creating_geometric_models/create_fault_from_trace.py

```diff
@@ -195,11 +193,8 @@ def generate_vertices(depth, sign=1):
 av0 = np.zeros((nx, 3))
 av1 = np.zeros((nx, 3))
 for i in range(0, nx):
-    if args.extrudeDir == None:
-        if i + 1 != nx:
-            v0 = nodes[i + 1, :] - nodes[i, :]
-        else:
-            v0 = nodes[i, :] - nodes[i - 1, :]
+    if not args.extrudeDir:
+        v0 = nodes[min(i + 1, nx-1), :] - nodes[max(i-1,0), :]
```

Comment:
Not quite sure, if this is correct:
if `i` is smaller than `nx` and larger than `1` (i.e. in the middle of the loop), then `min(i+1, nx-1) = i+1` and `max(i-1,0) = i-1`, but you never computed `nodes[i+1,:] - nodes[i-1,:]` before.
I guess this is your 5th bullet point.